src/DotNetWorker.Core/FunctionMetadata/DefaultFunctionMetadata.cs

```diff
@@ -20,15 +36,69 @@ public class DefaultFunctionMetadata : IFunctionMetadata
         public bool ManagedDependencyEnabled { get; set; }
 
         /// <inheritdoc/>
-        public string? Name { get; set; }
+        public string? Name { get => _name; set => ClearIdAndSet(value, ref _name); }
 
         /// <inheritdoc/>
-        public string? EntryPoint { get; set; }
+        public string? EntryPoint { get => _entryPoint; set => ClearIdAndSet(value, ref _entryPoint); }
 
         /// <inheritdoc/>
         public IList<string>? RawBindings { get; set; }
 
         /// <inheritdoc/>
-        public string? ScriptFile { get; set; }
+        public string? ScriptFile { get => _scriptFile; set => ClearIdAndSet(value, ref _scriptFile); }
+
+        private static string? HashFunctionId(DefaultFunctionMetadata function)
+        {
+            // we use uint to avoid the '-' sign when we .ToString() the result.
```

Comment:
I haven't used this before so thanks for adding the clarifying comment! Learned a new best practice today.

Reply:
I wouldn't say it's a best practice, just good to know what number type to use for the problem at hand. In this case, int and uint both have the same bits and some total number _range_, but just shifted across negatives or positives only:

``` CSharp
int: –2147483648 to 2147483647 
uint: 0 to 4294967295 
```